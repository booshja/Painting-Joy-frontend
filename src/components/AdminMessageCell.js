// dependencies
import React from "react";
import styled from "styled-components";

const StyledCell = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    border: 1px solid #207070;
    width: 100%;
    padding: 1rem;
    background-color: #ffffff;
`;

const StyledP = styled.p`
    font-size: 1.2rem;
    font-family: "News Cycle", sans-serif;
    letter-spacing: 1px;
    display: inline-block;
    word-break: break-word;
    margin-bottom: 0.5rem;
`;

const StyledBoldP = styled(StyledP)`
    font-weight: 700;
`;

const StyledOutlineButton = styled.button`
    border-radius: 4px;
    background-color: #ffffff;
    font-family: "News Cycle", sans-serif;
    font-weight: 700;
    font-size: 1.2rem;
    letter-spacing: 1px;
    padding: 2.5px 5px;
    border: 2px solid ${(props) => props.color};
    color: ${(props) => props.color};
`;

const StyledButtonContainer = styled.div`
    display: flex;
    justify-content: space-between;
    width: ${(props) => (props.showArchived ? "63%" : "55%")};
    align-self: flex-end;
    margin: 1rem 0 0;
`;

const AdminMessageCell = ({
    data,
    handleArchive,
    handleUnArchive,
    handleDelete,
    showArchived,
}) => {
    const { id, name, email, message } = data;
    return (
        <StyledCell>
            <StyledBoldP>{name}</StyledBoldP>
            <StyledP>{email}</StyledP>
            <StyledP>{message}</StyledP>
            <StyledButtonContainer showArchived={showArchived}>
                {showArchived ? (
                    <StyledOutlineButton
                        color="#a1c9c9"
                        onClick={() => handleUnArchive(id)}
                    >
                        Unarchive
                    </StyledOutlineButton>
                ) : (
                    <StyledOutlineButton
                        color="#a1c9c9"
                        onClick={() => handleArchive(id)}
                    >
                        Archive
                    </StyledOutlineButton>
                )}
                <StyledOutlineButton
                    color="#db9487"
                    onClick={() => handleDelete(id)}
                >
                    Delete
                </StyledOutlineButton>
            </StyledButtonContainer>
        </StyledCell>
    );
};

export default AdminMessageCell;
