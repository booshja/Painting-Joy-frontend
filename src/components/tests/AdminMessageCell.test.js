import { render } from "@testing-library/react";
import AdminMessageCell from "../AdminMessageCell";

/** Smoke Test */
it("renders without crashing, showArchived true", () => {
    render(
        <AdminMessageCell
            data={{
                id: 1,
                name: "John Doe",
                email: "johndoe@email.com",
                message: "This is a message from John Doe",
            }}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={true}
        />
    );
});

it("renders without crashing, showArchived false", () => {
    render(
        <AdminMessageCell
            data={{
                id: 1,
                name: "John Doe",
                email: "johndoe@email.com",
                message: "This is a message from John Doe",
            }}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={false}
        />
    );
});

/** Snapshot Test */
it("matches snapshot, showArchived true", () => {
    const { asFragment } = render(
        <AdminMessageCell
            data={{
                id: 1,
                name: "John Doe",
                email: "johndoe@email.com",
                message: "This is a message from John Doe",
            }}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={true}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, showArchived false", () => {
    const { asFragment } = render(
        <AdminMessageCell
            data={{
                id: 1,
                name: "John Doe",
                email: "johndoe@email.com",
                message: "This is a message from John Doe",
            }}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={false}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */

it("renders without crashing, showArchived true", () => {
    const { getByText, getByRole } = render(
        <AdminMessageCell
            data={{
                id: 1,
                name: "John Doe",
                email: "johndoe@email.com",
                message: "This is a message from John Doe",
            }}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={true}
        />
    );

    getByText("John Doe");
    getByText("johndoe@email.com");
    getByText("This is a message from John Doe");
    getByRole("button", { name: "Activate" });
    getByRole("button", { name: "Delete" });
});

it("renders without crashing, showArchived false", () => {
    const { getByText, getByRole } = render(
        <AdminMessageCell
            data={{
                id: 1,
                name: "Jane Doe",
                email: "janedoe@email.com",
                message: "This is a message from Jane Doe",
            }}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={false}
        />
    );

    getByText("Jane Doe");
    getByText("janedoe@email.com");
    getByText("This is a message from Jane Doe");
    getByRole("button", { name: "Archive" });
    getByRole("button", { name: "Delete" });
});
