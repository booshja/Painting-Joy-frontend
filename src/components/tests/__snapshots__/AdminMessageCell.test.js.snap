// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, showArchived false 1`] = `
<DocumentFragment>
  <div
    class="sc-hKFxyN kpxIQl"
  >
    <p
      class="sc-dlnjwi sc-eCApnc sc-jSFjdj dHRUYW dIiNo Mlpgf"
    >
      John Doe
    </p>
    <p
      class="sc-dlnjwi sc-eCApnc dHRUYW dIiNo"
    >
      <a
        href="mailto:johndoe@email.com"
      >
        johndoe@email.com
      </a>
    </p>
    <p
      class="sc-dlnjwi sc-eCApnc dHRUYW dIiNo"
    >
      This is a message from John Doe
    </p>
    <div
      class="sc-gKAaRy fbGlBn"
    >
      <button
        class="sc-gtsrHT qvdyu"
        color="#a1c9c9"
      >
        Archive
      </button>
      <button
        class="sc-gtsrHT blUuCa"
        color="#db9487"
      >
        Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, showArchived true 1`] = `
<DocumentFragment>
  <div
    class="sc-hKFxyN kpxIQl"
  >
    <p
      class="sc-dlnjwi sc-eCApnc sc-jSFjdj dHRUYW dIiNo Mlpgf"
    >
      John Doe
    </p>
    <p
      class="sc-dlnjwi sc-eCApnc dHRUYW dIiNo"
    >
      <a
        href="mailto:johndoe@email.com"
      >
        johndoe@email.com
      </a>
    </p>
    <p
      class="sc-dlnjwi sc-eCApnc dHRUYW dIiNo"
    >
      This is a message from John Doe
    </p>
    <div
      class="sc-gKAaRy fbGlBn"
    >
      <button
        class="sc-gtsrHT qvdyu"
        color="#a1c9c9"
      >
        Activate
      </button>
      <button
        class="sc-gtsrHT blUuCa"
        color="#db9487"
      >
        Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
